kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/util.kt

```diff
@@ -404,7 +405,10 @@ internal fun ClassId.toKtClassSymbol(): KaClassSymbol? {
                 it.asString() == this@toKtClassSymbol.shortClassName.asString()
             }?.singleOrNull() as? KaClassSymbol
         } else {
-            findClass(this@toKtClassSymbol)
+            // Try to find KaFirPsiJavaClassSymbol. See KT-74598 for details.
+            findClass(this@toKtClassSymbol)?.let {
+                (it.psi as? PsiClass)?.namedClassSymbol ?: it
```

Comment:
Instead of this unconditional trial, I wonder if you can check `origin` first, like `KaSymbolOrigin.JAVA_*`?

Reply:
Good point. Check of source is actually needed because psi can come from libraries 🤷